### ISSUE TITLE
Ask to select coding system when exit from Org mode

### ISSUE THREAD

**Issue by Eason0210** [purcell/emacs.d#717]
Hi Purcell,

I love to use your config on Haskell and Org. But when I exit from org mode to close emacs, it always ask me to select coding system. From the" temp file" buffer, org-clock-save.el inlude Chinese font. The issue is similar to issue #516 .  You can see the picture for detail.
 I tried to put  (set-language-environment 'UTF-8)
(set-locale-environment "UTF-8") in the init-local.el   and  put （ require 'init-local) in the init.el

But it still don't work.  Do you known how to set the coding system?
Thanks in advance.

![decoding](https://user-images.githubusercontent.com/47546452/59966026-b9dd9000-9548-11e9-92ae-9558e4790494.jpg)

**Comment by wisaruty** [comment]
เ เ คเ

ในวันที่ ส. 22 มิ.ย. 2019 22:53 Eason Huang <notifications@github.com>
เขียนว่า:

**Comment by purcell** [comment]
Yep, I don't think that's the same error: it looks like the code is already trying to use utf8, but that the characters are invalid. So perhaps there's a problem with how the characters get into the clock history, but I don't know. Perhaps it's worth asking about this on the `org` list?

**Comment by Eason0210** [comment]
Hi purcell，

Thanks for your your time and good suggestion. I will go to org list to find a solution.
Btw，I changed my calendar to English format，and everyting works well.

**Comment by Eason0210** [comment]
May be is Org-pomodoro cause this problem.  I put two lines of code in the init.el , It works fine.
(set-language-environment 'UTF-8)
(set-locale-environment "UTF-8")

**Comment by bloodhero** [comment]
i have same problem, and the method above work well for me, hope that commit to master branch

**Comment by purcell** [comment]
> i have same problem, and the method above works well for me, hope that commit to master branch

I feel like the best solution here is probably to properly set the system or user locale environment variables, e.g. `LC_CTYPE`, `LC_ALL`. This config tries not to force UTF-8 on every user, since I'm not sure that's best for all users, but if you set the env vars to indicate you want UTF-8, it will set up Emacs appropriately.

Ultimately, calling `set-local-environment` without passing a locale name will do the same thing, and I might switch the config to do this instead of the [current code](https://github.com/purcell/emacs.d/blob/master/lisp/init-locales.el).

**Comment by purcell** [comment]
Actually, correction, the config *does* force UTF-8 in GUI emacsen, so I'll probably simply switch to using `set-locale-environment`.

**Comment by purcell** [comment]
(Deleted a follow-up comment I made which didn't make sense.)

**Comment by Eason0210** [comment]
After this update, the issue is  still existed. but I can change my time format of Windows OS to fixed this issue.  For Chinese  Win 10 user with Chinese Language Interface, you can set your time format to English as follow:
![time-format](https://user-images.githubusercontent.com/47546452/75600859-a929e480-5aef-11ea-9cd9-a96a2e21ab54.jpg)

**Comment by shrubbroom** [comment]
@Eason0210 However this will prevent the lunar calendar from exhibiting in the Windows Calendar. Maybe solve this problem within Emacs is better.

**Comment by Eason0210** [comment]
> @Eason0210 However this will prevent the lunar calendar from exhibiting in the Windows Calendar. Maybe solve this problem within Emacs is better.
@shrubbroom  On my site, the lunar calendar is still there and work well.
Do you have better idear to solve this within Emacs? I can't find what kind of encoding system is using for the week showing in `org-clock-save.el`.
